spec/models/exercise_spec.rb

```diff
@@ -99,6 +99,14 @@ def set_deadline(ex, t)
     end
   end
 
+  def set_soft_deadline(ex, t)
+    if t.is_a? Array
+      ex.soft_deadline_spec = t.to_json
+    else
+      ex.soft_deadline_spec = [t.to_s].to_json
+    end
+  end
```

Comment:
I don't like the .is_a? -- could we maybe survive with flatten?
And what t is if it is not an array